modules/gapi/test/serialization/gsrlzkernel.cpp

```diff
@@ -0,0 +1,57 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#include "precomp.hpp"
+
+#include <cassert>
+
+#include "gsrlzkernel.hpp"
+
+const cv::Mat& cv::GSRLZContext::inMat(int input)
+{
+    return inArg<cv::Mat>(input);
+}
+
+cv::Mat&  cv::GSRLZContext::outMatR(int output)
+{
+    return *util::get<cv::Mat*>(m_results.at(output));
+}
+
+const cv::Scalar& cv::GSRLZContext::inVal(int input)
+{
+    return inArg<cv::Scalar>(input);
+}
+
+cv::Scalar& cv::GSRLZContext::outValR(int output)
+{
+    return *util::get<cv::Scalar*>(m_results.at(output));
+}
+
+cv::detail::VectorRef& cv::GSRLZContext::outVecRef(int output)
+{
+    return util::get<cv::detail::VectorRef>(m_results.at(output));
+}
+
+cv::detail::OpaqueRef& cv::GSRLZContext::outOpaqueRef(int output)
+{
+    return util::get<cv::detail::OpaqueRef>(m_results.at(output));
+}
+
+cv::GSRLZKernel::GSRLZKernel()
+{
+}
+
+cv::GSRLZKernel::GSRLZKernel(const GSRLZKernel::F &f)
+    : m_f(f)
+{
+}
+
+void cv::GSRLZKernel::apply(GSRLZContext &ctx)
+{
+    GAPI_Assert(m_f);
+    m_f(ctx);
+}
```

Comment:
Please mark this file for deletion. It means it won't be deleted in this review, but will be deleted as soon as we introduce proper tests for the serialization API (there will be no need in a backend and a kernel subsystem for that)

Reply:
???

Reply:
Can explain on a call